src/main/frontend/pipeline-console-view/pipeline-console/main/PipelineConsole.tsx

```diff
@@ -53,65 +62,70 @@ export default function PipelineConsole() {
         />
       </DropdownPortal>
 
-      <SplitView
-        direction={stageViewPosition === "top" ? "vertical" : "horizontal"}
-        storageKey="graph"
-      >
-        {(mainViewVisibility === "both" ||
-          mainViewVisibility === "graphOnly") &&
-          (loading ? (
-            <Skeleton />
-          ) : (
-            <Stages
-              stages={stages}
-              selectedStage={openStage || undefined}
-              stageViewPosition={stageViewPosition}
-              onStageSelect={handleStageSelect}
-            />
-          ))}
+      {showSplitView && (
+        <SplitView
+          direction={stageViewPosition === "top" ? "vertical" : "horizontal"}
+          storageKey="graph"
+        >
+          {!isOnlyPlaceholderNode &&
+            (mainViewVisibility === "both" ||
+              mainViewVisibility === "graphOnly") &&
+            (loading ? (
+              <Skeleton />
+            ) : (
+              <Stages
+                stages={stages}
+                selectedStage={openStage || undefined}
+                stageViewPosition={stageViewPosition}
+                onStageSelect={handleStageSelect}
+              />
+            ))}
+
+          <SplitView storageKey="stages" hide={isOnlyPlaceholderNode}>
+            {(mainViewVisibility === "both" ||
+              mainViewVisibility === "stagesOnly") && (
```

Comment:
rather than passing hide here would it be better to do something like
```suggestion
          <SplitView storageKey="stages>
            {((mainViewVisibility === "both" ||
              mainViewVisibility === "stagesOnly") && !isOnlyPlaceholderNode) && (
```